plugins/gpio/GPIODriver.cpp

```diff
@@ -168,7 +168,7 @@ bool GPIODriver::UpdateGPIOPins(const DmxBuffer &dmx) {
   };
 
   for (uint16_t i = 0;
-       i < m_gpio_pins.size() && (i + m_options.start_address < dmx.Size());
+       i < m_gpio_pins.size() && (i + m_options.start_address - 1u < dmx.Size());
```

Comment:
Looking at some other equivalent code in the SPI plugin, we calculate a `const unsigned int first_slot = m_options.start_address - 1;` then use that throughout, which should be slightly more optimal in the for loop, as well as clearer throughout generally. Would you mind making the same change to your PR?